### DISCUSSION TITLE
editable-columns: forceReloadAfterUpdate - Redraw entire table for some column updates, but not for others.

### DISCUSSION THREAD

**Discussion by tomsb** [Laravel-Backpack/community-forum#1231] [Ideas]
I've been working on improving the flexibility of how table updates are handled when certain columns are modified. In some cases, we want to reload the entire table to ensure everything stays consistent. However, for other columns, a full table redraw may not be necessary, and only the updated data needs to be refreshed. To address this, I propose adding a configurable `forceReloadAfterUpdate` feature.

There could be two ways to set `forceReloadAfterUpdate`:
```
protected function setupListOperation()
{
    // Reload after every update:
    CRUD::setOperationSetting('forceReloadAfterUpdate', true);
    
    // Or make a full table reload only if these fields are updated:
    CRUD::setOperationSetting('forceReloadAfterUpdate', [
        'column_name_1',
        'column_name_2',
    ]);
    
    // ...
}
```

Changes in `minor_update_client_logic.blade.php`
```diff
if (typeof forceReloadAfterUpdate === 'undefined') {
-    var forceReloadAfterUpdate = JSON.parse('{!! json_encode(\CRUD::getOperationSetting('forceReloadAfterUpdate') ?? false) !!}');
+   var forceReloadAfterUpdate = {{ var_export(\CRUD::getOperationSetting('forceReloadAfterUpdate') ?? false) }};
}

// ...

function completeRequest() {
-    if(forceReloadAfterUpdate) {
-        window.crud.table.ajax.reload();
-
-        return;
-    }
+    if(forceReloadAfterUpdate) {
+        if (typeof forceReloadAfterUpdate === 'boolean' && forceReloadAfterUpdate === true){
+            window.crud.table.ajax.reload();
+            return ;
+        }
+
+        if (typeof forceReloadAfterUpdate === 'object'){
+            if (forceReloadAfterUpdate.includes(columnName)){
+                window.crud.table.ajax.reload();
+            }
+        }
+        return;
+    }
    // Re-apply formatting to action column as dropdown
    if (typeof formatActionColumnAsDropdown !== 'undefined' && $('#crudTable').data('has-line-buttons-as-dropdown')) {
        formatActionColumnAsDropdown();
    }
}

// ...

// Update this in three places (in all three AJAX responses):
-   completeRequest();
+  completeRequest(container.dataset.columnName);
```

#### Conditional Reloading
A more advanced technique would be to update the `forceReloadAfterUpdate` variable after each AJAX request, allowing for a more granular approach. For instance, you might want to reload the entire table for `column_name_1` under certain conditions but not always. This can be managed through custom response rules/variables in `saveMinorUpdateEntry()`.

#### Row-Level Updates
Additionally, there could be another option to reload only specific table rows using `getMinorUpdateRow()`. For example:
Then there could be one more option to reload only specific table rows using `getMinorUpdateRow()`, for example:
```diff
protected function getMinorUpdateRow()
    {
       // ...
-     return CRUD::getEntriesAsJsonForDatatables([$this->minorUpdateEntry->withFakes()], 1, 1)['data'][0] ?? null;
+    $entries = [];
+    $entries[] = $this->minorUpdateEntry->withFakes();
+    $entries[] = $this->minorUpdateEntry->getMuCustomRelatedEntry()->withFakes(); // something like this
+    return CRUD::getEntriesAsJsonForDatatables($entries, count($entries), count($entries))['data'] ?? null;
```
This option would require a larger update in `minor_update_client_logic.blade.php` to identify the exact rows that need updating.

##
- Are there any concerns or suggestions for further refining this approach?
- Does the proposed configuration align well with your use cases?
- Any thoughts on implementing row-level updates?

[a comment by an automated account is not shown]

**Comment by karandatwani92** [comment]
Hey @tomsb 

Thanks for your suggestion and efforts. @pxpm, what do you think?

**Comment by pxpm** [comment]
Thanks @tomsb for the suggestion and for providing the code. 

I think the first suggestion is something that can be very useful, I've already created a PR from your code, did a small changes and released editable-columns 3.0.11 that provides this functionality. 

Regarding the second one, I am not sure about it, I think it may complicate the code a lot for a small benefit. We already load a limited set of entries in the table, so having more workarounds to just reload row X and Y is cumbersome IMO. 

I will be closing this now, please allow a few moments for our private repository to pick the changes and you should be able to get the new version with the changes you proposed. 

Nice idea, clean implementation, perfect 👍 

Thanks again @tomsb
